py/torch_tensorrt/dynamo/conversion/_ConverterRegistry.py

```diff
@@ -323,6 +329,18 @@ def __getitem__(
 
                 if isinstance(converters, (list, tuple)):
                     for candidate in converters:
+                        # TODO: Importing this here avoids circular import issue. One potential fix is moving this function into _ConverterRegistry file.
+                        from torch_tensorrt.dynamo.conversion.converter_utils import (
+                            dynamic_unsupported,
+                        )
+
+                        has_static_inputs = dynamic_unsupported(node)
+                        # If there are dynamic inputs but the converter doesn't support it explicitly, throw a warning.
+                        if not has_static_inputs and not candidate.dynamic:
```

Comment:
Switch to `if not candidate.dynamic and not dynamic_unsupported(node):`, to run the converter boolean check first and avoid the function call in most cases.